src/main/java/de/dennisguse/opentracks/io/file/importer/AbstractFileTrackImporter.java

```diff
@@ -127,21 +126,26 @@ public void characters(char[] ch, int start, int length) {
     }
 
     @Override
-    public Track.Id importFile(InputStream inputStream)  throws SAXException, SQLiteConstraintException, IOException, ParserConfigurationException {
+    public Track.Id importFile(InputStream inputStream) throws ImportParserException, ImportAlreadyExistsException {
         try {
             SAXParser saxParser = SAXParserFactory.newInstance().newSAXParser();
             long start = System.currentTimeMillis();
 
             saxParser.parse(inputStream, this);
             Log.d(TAG, "Total import time: " + (System.currentTimeMillis() - start) + "ms");
             if (trackIds.size() != 1) {
-                throw new MultiTracksImportException();
+                // TODO Multi track is not supported yet.
+                throw new ImportParserException("Multi track not supported");
             }
             return trackIds.get(0);
-        } catch (MultiTracksImportException | IOException | SAXException | ParserConfigurationException | SQLiteConstraintException e) {
+        } catch (IOException | SAXException | ParserConfigurationException e) {
             Log.e(TAG, "Unable to import file", e);
             cleanImport();
-            throw e;
+            throw new ImportParserException(e.getMessage());
```

Comment:
Put the whole Exception into the constructor not only the message. Then later you can call getCause() to get the original exception including Stack trace

Reply:
Okay :+1: